bundlewrap/secrets.py

```diff
@@ -97,35 +97,48 @@ def _decrypt(self, cryptotext=None, key='encrypt'):
 
         return Fernet(key).decrypt(cryptotext.encode('utf-8')).decode('utf-8')
 
-    def _decrypt_file(self, source_path=None, key='encrypt'):
+    def _decrypt_file(self, source_path=None, key=None):
         """
         Decrypts the file at source_path (relative to data/) and
         returns the plaintext as unicode.
         """
         if environ.get("BW_VAULT_DUMMY_MODE", "0") != "0":
             return "decrypted file"
-        try:
-            key = self.keys[key]
-        except KeyError:
-            raise FaultUnavailable(_(
-                "Key '{key}' not available for decryption of the following file, "
-                "check your {file}: {source_path}"
-            ).format(
-                file=FILENAME_SECRETS,
-                key=key,
-                source_path=source_path,
-            ))
+
+        key, content = self._determine_key_to_use(source_path=source_path, key=key)
 
         f = Fernet(key)
-        return f.decrypt(get_file_contents(join(self.repo.data_dir, source_path))).decode('utf-8')
+        return f.decrypt(content).decode('utf-8')
 
-    def _decrypt_file_as_base64(self, source_path=None, key='encrypt'):
+    def _decrypt_file_as_base64(self, source_path=None, key=None):
         """
         Decrypts the file at source_path (relative to data/) and
         returns the plaintext as base64.
         """
         if environ.get("BW_VAULT_DUMMY_MODE", "0") != "0":
             return b64encode("decrypted file as base64").decode('utf-8')
+
+        key, content = self._determine_key_to_use(source_path=source_path, key=key)
+
+        f = Fernet(key)
+        return b64encode(f.decrypt(content)).decode('utf-8')
+
+    def _determine_key_to_use(self, source_path=None, key=None):
+        content = get_file_contents(join(self.repo.data_dir, source_path))
+
+        key_delim = content.find(b'$')
+        if key_delim > -1:
+            key_from_file = content[:key_delim].decode('utf-8')
+            content = content[key_delim + 1:]
+        else:
+            key_from_file = None
+
+        if key is None:
```

Comment:
A key given as a parameter to this function wins over a key from the file. Why? Suppose decrypting the file using the key from the file fails. Now you want to test if another key works, so you use `bw debug -c "print(repo.vault.decrypt_file('test', key='hurz'))"`.